Automattic/wp-calypso
# Update storybook

#### Changes proposed in this Pull Request
Seeing #55736, I was motivated to try this cleanup ourselves.

Interestingly, storybook 6.4 has a lot of changes to prep for the future v7 release.
- Peer dependency issues are resolved, so we can remove our override. 🎉 
- It adds a flag to access modern emotion versions. This allows us to remove the hack which aliases them ourselves.
- There is a new `babelModeV7` feature. This opts-in to a feature which will be default in the next major update. Basically, Storybook is going to use your default babel config instead of whatever you pass directly. This allowed me to delete our custom babel stuff in the storybook config file.

However, there is one oddity:

```sh
Though the "loose" option was set to "false" in your @babel/preset-env config, it will not be used for @babel/plugin-proposal-private-property-in-object since the "loose" mode option was set to "true" for @babel/plugin-proposal-private-methods.
The "loose" option must be the same for @babel/plugin-proposal-class-properties, @babel/plugin-proposal-private-methods and @babel/plugin-proposal-private-property-in-object (when they are enabled): you can silence this warning by explicitly adding
	["@babel/plugin-proposal-private-property-in-object", { "loose": true }]
to the "plugins" section of your Babel config.
```

Now, we saw this same error previously, and I was able to fix it by adding `loose: false` for one of the babel plugins directly in the storybook config. However, **even if I use the config file from trunk**, this warning shows up. So it's not happening because of the new babel feature flag being enabled, or because of any of the other config changes.

From some research, the group of babel plugins related to class properties _should_ have loose set to `false`. This is the default set in `preset/env`. However, somewhere in our dependency tree, something is setting loose to true for one of the related plugins. I tried several things in our "default" babel preset in the babel config package we have. I wasn't able to find anything which worked, though, even setting all of the plugins to `loose: false`. (At some point, I was able to get the build to fail entirely with these changes, so Storybook is certainly reading our root babel config file now.)

I think we should continue without solving it for now, unless anyone has any good ideas :)

#### Testing instructions
Run the various storybooks locally, and make sure the components render. 

Side note: maybe we should just have one storybook project? It seems we are making new ones for each project, instead of adding all projects to the same one.
